tests/integration/components/select-component-test.js

```diff
@@ -133,3 +138,24 @@ test('It displays the specified component when componentNameForGroupTooltip is p
     'the groupItem argument is passed'
   );
 });
+
+test('It does not display loading text and empty content component at the same time', async function(assert) {
+  this.container.register('template:components/some-component', hbs`<span data-test-some-component>No results</span>`);
```

Comment:
Yeah I'm not sure this is enough mocking. As `emptyContentView` is used with `{{view` and not with the `{{component` helper I expect you need to register a view and template.